Geometry/TrackerNumberingBuilder/interface/GeometricDet.h

```diff
@@ -121,7 +121,8 @@ class GeometricDet {
   double phi() const { return _phi; }
   double rho() const { return _rho; }
 
-  DDSolidShape const& shape() const { return _shape; }
+  LegacySolidShape shape() const { return cms::dd::value(cms::LegacySolidShapeMap, _shape); }
+  cms::DDSolidShape const& shape_dd4hep() const { return _shape; }
```

Comment:
I don't see any point in returning a reference to an `enum`. The size of the reference and the `enum` are the same, so there's no saving of space. The method could just return the `enum`, like with `shape()`. But it's harmless.

Reply:
Primitive types are better returned by value instead of const reference, because reference is an indirection and the de-reference may incur a small penalty. In practice it may be that compiler can optimize it away, but nevertheless primitive types should be returned by value.